style/redux/README.md

```diff
@@ -0,0 +1,195 @@
+`createAction` is less readable and doesn't let us easily export named
+functions. Note that only `type` is mandatory.
+
+So far the common practice is to group files by type rather than by feature, and
+this style guide follows that pattern. But there's an alternative pattern -
+[ducks] - a proposal to bundle action creators with their reducers into a single
+file. Such a bundle is called a 'duck'. Its goal is to move towards a modular
+application structure.
+
+[redux-promise-middleware]: https://github.com/pburtchaell/redux-promise-middleware
+[redux-thunk]: https://github.com/gaearon/redux-thunk
+[Flux Standard Action]: https://github.com/acdlite/flux-standard-action
+[redux-actions]: https://github.com/acdlite/redux-actions
+[ducks]: https://github.com/erikras/ducks-modular-redux
+
+## Action types
+
+- Put action types in one file of constants.
+- Group related action types in an object.
+- Expose each object as a named export.
+- Expose all action types as the default export.
+
+```js
+// constants/actionTypes.js
+export const todo = {
+  fetch: 'FETCH_TODO',
+  set: 'SET_TODO',
+};
+
+export default {
+  todo,
+};
+```
+
+Action types identify the nature of the action that has occurred. Two actions
+with the same type MUST be strictly equivalent (using ===). By convention, type
+is usually a string constant or a Symbol.
+
+## Reducers
+
+- Expose reducers as the default export.
+- Expose handlers as named exports for unit testing.
+- Use constants instead of inline strings for action types.
+- Always define an `INITIAL_STATE` variable.
+
+```js
+// using redux-promise-middleware
+// using type-to-reducer
+
+// reducers/todo.js
+import typeToReducer from 'type-to-reducer';
+import { todo } from '../constants/actionTypes';
+
+const INITIAL_STATE = {
+  data: {},
+  error: false,
+  isRequesting: false,
+};
+
+export function addTodo(state, action) {
+  return { ...state, data: action.payload };
+}
+
+export const addTodoAsync = {
+  PENDING: (state) => ({
+    ...state,
+    isRequesting: true,
+  }),
+  REJECTED: (state, action) => ({
+    ...state,
+    isRequesting: false,
+    error: action.payload,
+  }),
+  FULFILLED: (state, action) => ({
+    ...state,
+    isRequesting: false,
+    data: action.payload,
+  }),
+};
+
+export default typeToReducer(
+  {
+    [todo.add]: addTodo,
+    [todo.addAsync]: addTodoAsync,
+  },
+  INITIAL_STATE
+);
+```
+
+The final reducer should be exposed as the default export. Individual handler
+functions should be exposed as named exports in order to simplify unit testing.
+
+The simplest way to create a reducer is to use the `handleActions` method of
+[redux-actions]. This avoids having to write a switch statement and a default
+handler. Another benefit is that it enforces the use of FSA-compliant action
+objects. Alternatively, use [type-to-reducer] for simplifying handling of async
+actions, as it allows nesting of related actions.
```

Comment:
There are other ways to write JavaScript, without so many ES6 tricks. For instance, this is a small piece of code I wrote recently.

```
import _ from 'lodash';
import { projectActionTypes, eventActionTypes } from '../constants';

const setEvent = (state, action) => {
  const newEvent = action.event;
  return state.map(event => {
    if (event.id === newEvent.id)
      return newEvent;
    return event;
  });
};

const setEvents = (state, action) => {
  const newIds = _.map(action.events, 'id');
  const selectedEvents = _.filter(state, event => !_.includes(newIds, event.id));
  return _.sortBy([...selectedEvents, ...action.events], 'id');
};

const addEvent = (state, action) => {
  return [...state, action.event];
};

export function events(state = [], action) {
  switch (action.type) {
    case projectActionTypes.ADD_EVENT:
      return addEvent(...arguments);
    case eventActionTypes.SET_EVENT:
      return setEvent(...arguments);
    case eventActionTypes.SET_EVENTS:
      return setEvents(...arguments);
    default:
      return state;
  }
}
```

There is more code, but it's clear what's happening: there is a function with a switch that calls other functions. It's reusable and clear in my opinion.